back-end/service.py

```diff
@@ -123,62 +136,66 @@ def updateSecret(secret, data, user, repo):
         })
 
     if res.status_code == 200:
-        log('info', 'Secret updated.')
+        log('debug', 'Secret updated in `{}`'.format(repo))
         return True
     else:
-        log('error', 'Secret could not be updated.')
+        log('warning', 'Secret could not be updated in `{}`'.format(repo))
         log('critical', 'Execution may not work properly from here.')
         return False
 
 
-def ensureSecret(user, repo):
+def ensureSecrets(user):
 
     repos = requests.get(
         DRONE_ADDR + "/api/user/repos",
         headers={
             'Authorization': 'Bearer {}'.format(os.environ['DRONE_TOKEN'])
             }).json()
 
-    if repo not in [x['name'] for x in repos if x['active']]:
-        
-        log('error', 'Repo {} is not activated in Drone.'.format(repo))
+    for repo in repos:
+        if not repo['active']:
```

Comment:
Does the service activate all inactive repositories on login to add secrets?

Reply:
Yes.

Reply:
It should only rekey already activated ones.  Repositories that the user hasn't activated shouldn't be added for no reason.